osu.Game/Database/ArchiveModelManager.cs

```diff
@@ -212,6 +218,27 @@ public TModel Import(ArchiveReader archive)
             }
         }
 
+        /// <summary>
+        /// Any file extensions which should be included in hash creation.
+        /// Generally should include all file types which determine the file's uniqueness.
+        /// Large files should be avoided if possible.
+        /// </summary>
+        protected abstract string[] HashableFileTypes { get; }
+
+        /// <summary>
+        /// Create a SHA-2 hash from the provided archive based on contained beatmap (.osu) file content.
+        /// </summary>
+        private string computeBeatmapSetHash(ArchiveReader reader)
+        {
+            // for now, concatenate all .osu files in the set to create a unique hash.
+            MemoryStream hashable = new MemoryStream();
+            foreach (string file in reader.Filenames.Where(f => f.EndsWith(".osu")))
```

Comment:
Was this tested at all? The `HashableFileTypes` property isn't even used.

Reply:
It actually did work while testing, but probably because I only tested for one skin.